common/meparser.cpp

```diff
@@ -54,7 +53,7 @@ USTATUS MeParser::parseMeRegionBody(const UModelIndex & index)
     
     // Check ME signature to determine it's version
     // ME v11 and older layout
-    if (meRegion.left(sizeof(UINT32)) == FPT_HEADER_SIGNATURE || meRegion.mid(ME_ROM_BYPASS_VECTOR_SIZE, sizeof(UINT32)) == FPT_HEADER_SIGNATURE) {
+    if (*(UINT32*)meRegion.constData() == FPT_HEADER_SIGNATURE || *(UINT32*)(meRegion.constData() + ME_ROM_BYPASS_VECTOR_SIZE)  == FPT_HEADER_SIGNATURE) {
```

Comment:
Is this guaranteed to have 4 bytes?

Reply:
Yes, but it needs to be replaced with readUnaligned, because all this `*(UINTX)constCharPtr` shit breaks strict aliasing anyway. I will make a round of refactoring specifically addressing that later.